src/components/flame-graph/FlameGraph.js

```diff
@@ -171,13 +181,8 @@ class FlameGraph extends React.PureComponent<Props> {
         break;
       }
       case 'ArrowUp': {
-        const children = callTree.getChildren(selectedCallNodeIndex);
-        // The order of the children returned from getChildren does
-        // not correspond to the paint order in the flame graph, so
-        // the one chosen for selection might look rather arbitrary.
-        const callNodeIndex = children.find(this._wideEnough);
-
-        if (callNodeIndex !== undefined) {
+        const [callNodeIndex] = callTree.getChildren(selectedCallNodeIndex);
```

Comment:
nit: still maybe it's worth explaining in a comment that the children are sorted by time length :)